### PR TITLE
Tudor/authenticated contract id 3x

### PR DESCRIPTION
**Pull request by tudor-da**

### CODE REVIEW

File: canton/community/base/src/main/scala/com/digitalasset/canton/protocol/SerializableRawContractInstance.scala [hunk 1]
```diff
@@ -45,6 +45,7 @@ final case class SerializableRawContractInstance private (
 
   lazy val contractHash: LfHash =
     LfHash.assertHashContractInstance(
+      contractInstance.unversioned.packageName,
```
Comment (tudor-da) [inline canton/community/base/src/main/scala/com/digitalasset/canton/protocol/SerializableRawContractInstance.scala:48]: This and the related adaptation to `LfHash.assertHashContractInstance` are the gist of the changes for the new contract id authentication scheme in 3.x. @remyhaemmerle-da @andreaslochbihler-da

File: canton/community/base/src/main/scala/com/digitalasset/canton/logging/pretty/PrettyInstances.scala [hunk 2]
```diff
@@ -222,7 +222,7 @@ trait PrettyInstances {
   implicit def prettyLfContractId: Pretty[LfContractId] = prettyOfString {
     case LfContractId.V1(discriminator, suffix)
         // Shorten only Canton contract ids
-        if suffix.startsWith(AuthenticatedContractIdVersionV2.versionPrefixBytes) =>
+        if suffix.startsWith(AuthenticatedContractIdVersionV3.versionPrefixBytes) =>
```
Comment (andreaslochbihler-da) [inline canton/community/base/src/main/scala/com/digitalasset/canton/logging/pretty/PrettyInstances.scala:225]: You probably want to shorten V2 and V3 going forward.

**Review by andreaslochbihler-da — COMMENTED** [review]
The commit 07a45d9cd75c083541736d45faa1318fe51ed8a4 looks good. On `main`, it's OK to replace V2 with V3. On `main-2.x`, you will want to keep all the existing Canton contract ID versions.

**Comment by tudor-da** [comment]
@andreaslochbihler-da Thanks for the input. Closing this as superseeded by:
* https://github.com/digital-asset/daml/pull/18506
* https://github.com/DACH-NY/canton/pull/17202

**Pull request closed without merging**